server.go.tmpl

```diff
@@ -66,68 +74,51 @@ func (s *{{$serviceName}}) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 {{range .Methods }}
 func (s *{{$serviceName}}) serve{{ .Name | firstLetterToUpper }}JSON(ctx context.Context, w http.ResponseWriter, r *http.Request) {
-	var err error
-	ctx = context.WithValue(ctx, MethodNameCtxKey, "{{.Name}}")
+	{{ if .Inputs|len}}
+	reqBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		RespondWithError(w, ErrorWithCause(ErrWebrpcBadRequest, fmt.Errorf("failed to read request data: %w", err)))
+		return
+	}
+	defer r.Body.Close()
 
-	{{- if .Inputs|len}}
-	reqContent := struct {
+	reqPayload := struct {
 	{{- range $i, $input := .Inputs}}
 		Arg{{$i}} {{template "field" dict "Name" $input.Name "Type" $input.Type "Optional" $input.Optional "TypeMap" $typeMap "TypePrefix" $typePrefix "TypeMeta" $input.Meta "JsonTags" true}}
 	{{- end}}
 	}{}
-
-	reqBody, err := io.ReadAll(r.Body)
-	if err != nil {
-		err = ErrorWithCause(ErrWebrpcBadRequest, fmt.Errorf("failed to read request data: %w", err))
-		RespondWithError(w, err)
+	if err := json.Unmarshal(reqBody, &reqPayload); err != nil {
+		RespondWithError(w, ErrorWithCause(ErrWebrpcBadRequest, fmt.Errorf("failed to unmarshal request data: %w", err)))
 		return
 	}
-	defer r.Body.Close()
+	{{- end}}
 
-	err = json.Unmarshal(reqBody, &reqContent)
+	ctx = context.WithValue(ctx, MethodNameCtxKey, "{{.Name}}")
+
+	// Call service method implementation.
+	{{range $i, $output := .Outputs}}ret{{$i}}, {{end}}err := s.{{$name}}.{{.Name}}(ctx{{range $i, $_ := .Inputs}}, reqPayload.Arg{{$i}}{{end}})
```

Comment:
Move the service call to the main handler function. Remove the unnecessary defer and anonymous function.